src/sgl/device/device.cpp

```
@@ -1372,6 +1409,77 @@ void Device::_unregister_device_child(DeviceChild* device_child)
+    CommandBuffer* command_buffer,
+    uint64_t submit_id
+)
+{
+    CommandRecordingSubmittedEvent event{
+        .device = this,
+        .id = id,
+        .command_buffer = command_buffer,
+        .submit_id = submit_id,
+    };
+    for (const auto& [_, callback] : m_command_recording_submitted_callbacks)
+        callback(event);
```

Cppcheck (2.20.0)

[error] Found an exit path from function with non-void return type that has missing return statement

(missingReturn)